hibernate-types-52/src/test/java/com/vladmihalcea/hibernate/util/providers/MySQLDataSourceProvider.java

```diff
@@ -1,73 +1,23 @@
 package com.vladmihalcea.hibernate.util.providers;
 
 import com.mysql.cj.jdbc.MysqlDataSource;
+import org.testcontainers.containers.MySQLContainer;
 
 import javax.sql.DataSource;
-import java.util.Properties;
 
 /**
  * @author Vlad Mihalcea
  */
 public class MySQLDataSourceProvider implements DataSourceProvider {
+    private static final MySQLContainer<?> MYSQL_CONTAINER;
 
-    private boolean rewriteBatchedStatements = true;
+    static {
+        MYSQL_CONTAINER = new MySQLContainer<>("mysql:5.7")
```

Comment:
I wouldn't start the container unless we cannot find the DB using the default settings. So, I'd use lazy initialization for it at bootstrap time. If we get a connection failure, then we bootstrap the Testcontainers DB.